tests/test_function.py

```diff
@@ -0,0 +1,102 @@
+import pytest
+import numpy as np
+
+from devito import Dimension, Function, Grid, Eq, Operator
+from devito.data.allocators import DataReference
+
+
+class TestRebuild:
```

Comment:
I don't think we need a new file for this...

this class can go inside test_symbolics ...

Reply:
`test_symbolics` is a mess which really needs tidying up. It's currently an unordered grab bag of miscellaneous tests. Maybe that should be done in this PR?

Reply:
I can do with a test_rebuild.py in which we gather this TestRebuild and all of the rebuilding tests in test_symbolics.py

Reply:
Yeah, that's a solution